File: tbsmith/day3.php.com/views/helpers/form_validator.php
<?php

Class FormValidator {

    function validate($query) {

        $data['successful'] = true;

        // name
        if (!preg_match('/^[a-zA-Z ]+$/', $query['name'])) {
            $data['successful'] = false;
            $data['name']['has_error'] = true;
            $data['name']['error_message'] = "Name must contain only letters and spaces.";
        }   

        // email
        if (!preg_match('/^\w+@\w+\.\w+$/', $query['email'])) {
            $data['successful'] = false;
            $data['email']['has_error'] = true;
            $data['email']['error_message'] = "Email address must look like user@example.com.";
        }   

        // verify email
        if ($query['email'] != $query['verifyemail']) {
            $data['successful'] = false;
            $data['verifyemail']['has_error'] = true;
            $data['verifyemail']['error_message'] = "The email addresses did not match.";
        }   
        
        // password
        if (strlen($query['password']) < 8) {
            $data['successful'] = false;
            $data['password']['has_error'] = true;
            $data['password']['error_message'] = "The password must be at least 8 characters long.";
        }   

        //rating
        if (empty($query['rating'])) {
            $data['successful'] = false;
            $data['rating']['has_error'] = true;
            $data['rating']['error_message'] = "Rating must be set.";
        }   
    
        //fav number
        if (!preg_match('/^[0-9]+$/', $query['favNumber'])) {
            $data['successful'] = false;
            $data['favNumber']['has_error'] = true;
            $data['favNumber']['error_message'] = "Favnumber must be an integer.";
        }   

        return $data;

    }   

}

?>

File: tbsmith/day3.python.com/views/helpers/form_validator.py
#!/usr/bin/python

import re


class FormValidator():

    def __init__(self):
        self.data = {}
        self.data['name'] = {}
        self.data['email'] = {}
        self.data['verifyemail'] = {}
        self.data['password'] = {}
        self.data['rating'] = {}
        self.data['favNumber'] = {}

    def validate(self, query):
        self.data['successful'] = True

        # name
        if 'name' not in query or \
           re.search('^[a-zA-Z ]+$', query.getvalue('name')) is None:
            self.data['successful'] = False
            self.data['name']['has_error'] = True
            self.data['name']['error_message'] = \
                "Name must contain only letters and spaces."

        # email
        if 'email' not in query or \
           re.search('^\w+@\w+\.\w+', query.getvalue('email')) is None:
            self.data['successful'] = False
            self.data['email']['has_error'] = True
            self.data['email']['error_message'] = \
                "Email must be like user@example.com."

        # verify email
        if 'verifyemail' not in query or \
           query.getvalue('email') != query.getvalue('verifyemail'):
            self.data['successful'] = False
            self.data['verifyemail']['has_error'] = True
            self.data['verifyemail']['error_message'] = \
                "Email addresses must match."

        # password
        if 'password' not in query or \
           len(query.getvalue('password')) < 8:
            self.data['successful'] = False
            self.data['password']['has_error'] = True
            self.data['password']['error_message'] = \
                "Password must be at least 8 characters long."

        # rating
        if 'rating' not in query:
            self.data['successful'] = False
            self.data['rating']['has_error'] = True
            self.data['rating']['error_message'] = "Rating must be set."

        # favNumber
        if re.search('^\d+$', query.getvalue('favNumber')) is None:
            self.data['successful'] = False
            self.data['favNumber']['has_error'] = True
            self.data['favNumber']['error_message'] = \
                "FavNumber must be an integer."

        return self.data
